src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -392,10 +501,81 @@ def _validate_vmss_create_load_balancer(namespace):
 
 def process_vmss_create_namespace(namespace):
     validate_location(namespace)
-    _validate_vm_create_storage_profile(namespace)
+    _validate_vm_create_storage_profile(namespace, for_scale_set=True)
     _validate_vmss_create_load_balancer(namespace)
     _validate_vm_create_vnet(namespace)
-    _validate_vm_create_public_ip(namespace)
+    _validate_vmss_create_subnet(namespace)
+    _validate_vmss_create_public_ip(namespace)
     _validate_vm_create_auth(namespace)
 
 # endregion
+
+# region disk, snapshot, image validators
+
+
+def validate_vm_disk(namespace):
+    namespace.disk = _get_ressource_id(namespace.disk, namespace.resource_group_name,
+                                       'disks', 'Microsoft.Compute')
+
+
+def process_disk_or_snapshot_create_namespace(namespace):
+    temp = [namespace.source_disk, namespace.source_snapshot, namespace.source_blob_uri]
+    if sum([1 for t in temp if t]) > 1:
+        raise CLIError('Only one source can be specified: --source-disk | --source-snapshot | '
+                       '--source-blob-uri')
```

Comment:
This is another reason why we should only have `--source` and then determine here whether it is a disk, snapshot or uri.